inject.js

```diff
@@ -415,27 +439,46 @@
         // Process the DOM nodes lazily
         requestIdleCallback(_ => {
           mutations.forEach(function(mutation) {
-            forEach.call(mutation.addedNodes, function(node) {
-              if (typeof node === "function")
-                return;
-              checkForVideo(node, node.parentNode || mutation.target, true);
-            });
-            forEach.call(mutation.removedNodes, function(node) {
-              if (typeof node === "function")
-                return;
-              checkForVideo(node, node.parentNode || mutation.target, false);
-            });
+            switch (mutation.type) {
+              case 'childList':
+                forEach.call(mutation.addedNodes, function(node) {
+                  if (typeof node === "function")
+                    return;
+                  checkForVideo(node, node.parentNode || mutation.target, true);
+                });
+                forEach.call(mutation.removedNodes, function(node) {
+                  if (typeof node === "function")
+                    return;
+                  checkForVideo(node, node.parentNode || mutation.target, false);
+                });
+                break;
+              case 'attributes':
+                if ((mutation.target.tagName == 'APPLE-TV-PLUS-PLAYER') && (mutation.target.attributes['aria-hidden'].value == "false")) {
```

Comment:
there are now only two references to apple in the inject.js. much less than before

Reply:
If I'm reading this correctly, you're checking that the aria-hidden is now `true` but also gating it on apple-tv-plus-player. Correct? Any reason why we can't remove apple-tv requirement? Per the discussion below, we can ignore hidden elements — that makes sense.

Reply:
> Correct?

that is correct, @igrigorik.

certain elements like the login modal also have their aria-hidden value change to `true`. 
I added the check for the tag name first so that it will short-circuit and not check the next statement. 

this is maybe the third time that you've mentioned removing the apple-tv in the code? I'm going to assume that you probably don't hate apple 🤣, but actually you don't want very many site specific references in this so that the codebase will be easier to maintain and generalizable to different websites that use shadow roots. the only part of this PR that definitely needs an apple-tv specific reference is on the section around 
```js
case (location.hostname == 'tv.apple.com'):
```
maybe it needs to be changed to also include music.apple.com since there might be music videos. 

### ideas

maybe a different mutation observer would need to be created each time there is a shadow root attachment. I don't see any classes here, and I'm not sure how to do that with another function attached inside the mutation observer. it might be easier with classes, and if you make use of classes, you might as well change the vsc controller to be a web component class to make your codebase seem hip and up to date with the latest standards. but i digress. 

it might be better to wait for v2 of the shadowDOM to come out and a new mutation observer spec to be published rather than trying to shoehorn a shadowRoot mutationObserver into this codebase. do you have any sway with WHATWG? because they have been shooting down this idea more than once a couple years ago, but i don't know if they're open to reevaluating their position.

### 2020
this is probably not the best place to put this comment in, but what is the roadmap for the next year in terms of updating the codebase to make it seem more modern?

Reply:
Indeed, sorry if I'm being repetitive, but my main question here is whether we can generalize this codepath to be not Apple specific (nothing against Apple, to the contrary :)). Specifically, I was referring to this clause in the if..

`mutation.target.tagName == 'APPLE-TV-PLUS-PLAYER'`

I think we can drop that and everything should continue to work?

Re, modernize: definitely open to it! Want to start a separate issue with more details on what you have in mind?

Reply:
> I think we can drop that and everything should continue to work?

yes. i added in another check to make sure it finds a video tag before proceeding. this check is just to make sure that the subsequent statements don't fail on aria-hidden=false elements that appear without a video tag, e.g. login modals (but apple uses iframes for the login modal, rather than shadow roots).

Reply:
> Re, modernize:
yes i'll do that.